pkg/operator/ceph/object/spec.go

```diff
@@ -253,7 +276,29 @@ func (c *clusterConfig) vaultTokenInitContainer(rgwConfig *rgwConfig) v1.Contain
 		SecurityContext: controller.PodSecurityContext(),
 	}
 }
-
+func (c *clusterConfig) vaultAgentContainer(rgwConfig *rgwConfig) v1.Container {
+	vaultAgentVolMount := v1.VolumeMount{
+		Name:      "vault-agent-config",
+		ReadOnly:  true,
+		MountPath: "/etc/vault/agent/",
+	}
+	return v1.Container{
+		Name:  "vault-agent-container",
+		Image: "vault",
+		Command: []string{
+			"vault",
+		},
+		Args: []string{
+			"agent",
+			"-config=/etc/vault/agent/vault-agent-config.hcl",
+			"-log-level=debug",
+		},
+		Ports:           []v1.ContainerPort{{ContainerPort: 8100}},
+		VolumeMounts:    []v1.VolumeMount{vaultAgentVolMount},
+		Resources:       c.store.Spec.Gateway.Resources,
```

Comment:
@leseb do you know if this is going to double the resource requests limits of the RGW pod?